.github/workflows/powershell-analyzer.yml

```diff
@@ -36,14 +36,14 @@ jobs:
         with:
           directory: ./testing-repo/
 
-      - name: Upload Result
-        uses: github/codeql-action/upload-sarif@1fae5bf71b0ecdc7d0a2ef0d0c28409d99693966 #v2.9.2
-        with:
-          sarif_file: ./results.sarif
-
-      - name: Upload SARIF Result
+      - name: Upload SARIF as Artifact
         uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
         with:
             name: analysis-result
             path: ./results.sarif
             retention-days: 10
+
+      - name: Upload SARIF to CodeQL
+        uses: github/codeql-action/upload-sarif@1fae5bf71b0ecdc7d0a2ef0d0c28409d99693966 #v2.9.2
+        with:
+          sarif_file: ./results.sarif
```

Comment:
I ran into a situation where analyze.ps1 did not have a UTF byte order mark. The Sarif file had `lineNumber: null` and the upload to CodeQL failed. `null` makes sense, given that the BOM is literally before any other lines. But I couldn't figure out what was going on until I could create the sarif file on my own and read through it. Having it upload to Artifacts before trying to upload to CodeQL will make this kind of diagnosis easier in the future.